src/player.ts

```diff
@@ -96,209 +74,48 @@ export class Player {
     }
   }
 
-  private attack() {
-    this.isAttacking = true;
-    this.attackTimer = Date.now();
-  }
-
-  private updateAttack(enemies: Array<Enemy>) {
-    const elapsed = Date.now() - this.attackTimer;
-    if (elapsed >= this.weapon.attackDuration) {
-      this.isAttacking = false;
-    }
-    const weaponRotation =
-      this.facing +
-      this.weapon.swingAngle / 2 -
-      this.weapon.swingAngle *
-        ((Date.now() - this.attackTimer) / this.weapon.attackDuration);
-
-    const weaponCollision = collision.weapon(
-      this.weapon,
-      weaponRotation,
-      this.x,
-      this.y,
-      this.width / 2,
-      enemies
-    );
-    if (weaponCollision.length > 0) {
-      for (let i = 0; i < weaponCollision.length; i++) {
-        enemies.splice(weaponCollision[i], 1);
-      }
-    }
-  }
-
   draw() {
-    if (this.context === null) {
+    if (this.ctx === null) {
       throw new Error('CanvasRenderingContext2D is null.');
     }
-
-    this.context.save();
-    this.context.translate(this.x, this.y);
-    this.context.rotate(this.facing);
-
-    this.context.fillStyle = this.color;
-    this.context.fillRect(
-      (-1 * this.width) / 2,
-      (-1 * this.height) / 2,
-      this.width,
-      this.height
+    const body = this.body;
```

Comment:
I would just use this.body throughout the code.

Reply:
done